test/unit/warp/src/Shfl.cpp

```diff
@@ -0,0 +1,127 @@
+/* Copyright 2021 David M. Rogers
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <alpaka/test/KernelExecutionFixture.hpp>
+#include <alpaka/test/acc/TestAccs.hpp>
+#include <alpaka/test/queue/Queue.hpp>
+#include <alpaka/warp/Traits.hpp>
+
+#include <catch2/catch.hpp>
+
+#include <cstdint>
+
+//#############################################################################
+class ShflSingleThreadWarpTestKernel
+{
+public:
+    //-------------------------------------------------------------------------
+    ALPAKA_NO_HOST_ACC_WARNING
+    template<typename TAcc>
+    ALPAKA_FN_ACC auto operator()(TAcc const& acc, bool* success) const -> void
+    {
+        std::int32_t const warpExtent = alpaka::warp::getSize(acc);
+        ALPAKA_CHECK(*success, warpExtent == 1);
+
+        ALPAKA_CHECK(*success, alpaka::warp::shfl(acc, 12, 0) == 12);
+        ALPAKA_CHECK(*success, alpaka::warp::shfl(acc, 42, -1) == 42);
+        float ans = alpaka::warp::shfl(acc, 3.3f, 0);
+        ALPAKA_CHECK(*success, alpaka::math::abs(acc, ans - 3.3f) < 1e-8f);
```

Comment:
Not the solution I had in mind, but LGTM!

Reply:
Could we define a misc function in [this folder](https://github.com/alpaka-group/alpaka/tree/develop/test/common/include/alpaka/test) and use the function here?.

```C++
template<typename TAcc, typename T>
bool equal(TAcc const & acc, T const & v0, T const & v1, T const empsilon = std::numeric_limits<T>::epsilon() )
{
  return alpaka::math::abs(acc, v0 - v1) < empisilon;
}
```

Reply:
You could also use `std::numeric_limits<float>::epsilon()` directly instead of the magic number `1.e-8`
This woild be simpler then including the helper function.

Reply:
I pushed a fix to this PR which used `std::numeric_limits::epsilon()`